source/framework/core/src/TRestProcessRunner.cxx

```diff
@@ -211,14 +223,18 @@ Int_t TRestProcessRunner::ReadConfig(string keydeclare, TiXmlElement* e) {
                     fRunInfo->SetExtProcess(p);
                     return 0;
                 }
-                if (fThreadNumber > 1 && (p->GetVerboseLevel() >= REST_Debug || p->singleThreadOnly())) {
+                if ((p->GetVerboseLevel() >= REST_Debug || p->singleThreadOnly())) {
+                    fUsePauseMenu = false;
                     fProcStatus = kIgnore;
-                    info << "multi-threading is disabled due to process \"" << p->GetName() << "\"" << endl;
-                    info << "This process is in debug mode or is single thread only" << endl;
-                    for (i = fThreadNumber; i > 1; i--) {
-                        delete (*(fThreads.end() - 1));
-                        fThreads.erase(fThreads.end() - 1);
-                        fThreadNumber--;
+                    if (fThreadNumber > 1) {
+                        info << "multi-threading is disabled due to process \"" << p->GetName() << "\""
+                             << endl;
+                        info << "This process is in debug mode or is single thread only" << endl;
+                        for (i = fThreadNumber; i > 1; i--) {
+                            // delete (*fThreads.end());
+                            fThreads.erase(fThreads.end() - 1);
+                            fThreadNumber--;
+                        }
```

Comment:
Don't quite understand this is this bucle for? Note that you have to `delete` the pointer content, if not you have a potential memory leak.
```suggestion
                        for (auto it = fThread.begin(); it != fThread.end(); ) {
                           if( it !=fThread.begin() ) {
                             delete * it;
                             it = fThread.erase(it);
                             fThreadNumber--;
                           } else {
                             ++it;
                           }
                        }
```

Reply:
I don't know why but it always causes problems when I delete the TRestThread object. This code block is for a rare case that, when adding processes, one process turns out to be must running under single thread. So we remove other threads. I guess this memory leak is acceptable and so I directly comment the delete operation.

Reply:
But it was deleted in the previous implementation...

Reply:
In the previous implementation this delete operation sometimes causes seg. fault which I don't know why.

Reply:
Now that I check more thoroughly, looks like you have a double bucle for `fThreads`, which can be the origin of the issue.
```
for (int i = 0; i < fThreadNumber; i++) {
....
               for (i = fThreadNumber; i > 1; i--)
```
I would suggest to use std iterators instead:
```
for (auto it = fThread.begin(); it != fThread.end(); ) {
```